ipalib/cli.py

```diff
@@ -1454,10 +1454,48 @@ def format_description(self, description):
 )
 
 
+def acquire_cred_interactively(*args, **kw):
```

Comment:
The function should never receive positional arguments and always the keyword arguments api, principal, and ccache, I recommend this syntax:
```suggestion
def acquire_cred_interactively(*, api, principal, ccache, **kwargs):
```

Reply:
Thanks, makes sense. It is a nice feature.
I updated the code and also decided to pass in an exception class to raise in case the callback decides it cannot acquire credentials. This helps against loops within `self.get_auth_info()`: `self._handle_exception()` -> `self._try_acquire_cred()` -> `raise errors.NonFatalError()` -> `self.get_auth_info()`.